MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,778 @@
+    @inject(GLSPActionDispatcher)
+    protected actionDispatcher: GLSPActionDispatcher;
+
+    postModelInitialization(): MaybePromise<void> {
+        this.actionDispatcher.dispatch(SetUIExtensionVisibilityAction.create({ extensionId: 'my-extension', visible: true }));
+    }
+}
+
+
+export const myDiagramModule= new ContainerModule(bind => {
+    bind(TYPES.IDiagramStartup).to(MyDiagramStartup).inSingletonScope();
+}
+```
+
+### Event Listener
+
+With 2.x a generic Event Emitter API has been introduced.
+This API is similar to the event API of Theia/VS Code and allows declaring arbitrary events and corresponding Emitters.
+Interesting parties can subscribe to the event to get notified when the event is fired.
+The existing listener implementations for `SelectionService`, `GLSPCommandStack` & `EditorContextService` have been refactored to use the new API.
+Explicit methods for de-registering listeners are no longer available.
+Instead adopters have to keep track of the disposable that was received when registering the listener and dispose it when the listener is no longer needed.
+
+<details>
+  <summary>1.0.0</summary>
+
+```ts
+@injectable()
+export class MyService {
+    @inject(SelectionService)
+    protected selectionService: SelectionService;
+
+    protected listener: SelectionListener;
+
+    @postConstruct()
+    protected init(): void {
+        this.listener = {
+            selectionChanged: () => {
+                console.log('selection changed');
+            }
+        };
+        this.selectionService.register({
+            selectionChanged: () => {
+                console.log('selection changed');
+            }
+        });
+    }
+
+    dispose(): void {
+        this.selectionService.deregister(this.listener);
+    }
+}
+```
+
+</details>
+<details open>
+  <summary>2.x</summary>
+
+```ts
+@injectable()
+export class MyService implements Disposable {
+    @inject(SelectionService)
+    protected selectionService: SelectionService;
+
+    protected selectionListenerDisposable: Disposable;
+
+    @postConstruct()
+    protected init(): void {
+        this.selectionListenerDisposable = this.selectionService.onSelectionChanged(() => {
+            console.log('selection changed');
+        });
+    }
+
+    dispose(): void {
+        this.selectionListenerDisposable.dispose();
+    }
+}
+```
+
+<details>
+  <summary>List of changes</summary>
+
+-   SelectionListener -> ISelectionListener
+-   EditModeListener -> IEditModeListener
+
+</details>
+
+### Message API
+
+With 2.x the actions for sending information messages and/or status updates have been reworked.
+First of all they have been renamed
+
+-   ServerStatusAction -> StatusAction
+-   ServerMessageAction -> MessageAction
+
+Second, the `timeout` support for `MessageAction`'s has been removed.
+This feature was only well supported in the Theia integration and its main purpose was to dispatch progress information for long-running operations.
```

Comment:
Customizations may have taken advantage of that timeout, e.g., in a toast notification or something similar. Still it was removed, maybe we can soften the sentence and just say that `By default, this feature was only...`?